AwsEncryptionSDK/runtimes/net/Examples/Keyring/AwsKmsHierarchical/AwsKmsHierarchicalKeyring.cs

```diff
@@ -34,175 +37,170 @@
+///
 /// This example also requires using a KMS Key.
 /// You need the following access on this key:
 /// - GenerateDataKeyWithoutPlaintext
+/// - ReEncrypt
 /// - Decrypt
 /// </summary>
 public class AwsKmsHierarchicalKeyring
 {
-    // THESE ARE PUBLIC RESOURCES DO NOT USE IN A PRODUCTION ENVIRONMENT
-    private static string branchKeyIdA = "43574aa0-de30-424e-bad4-0b06f6e89478";
-    private static string branchKeyIdB = "a2f4be37-15ec-489a-bcb5-dcce1f6bfe84";
-    private static void Run(MemoryStream plaintext)
+    private void Run(MemoryStream plaintext)
     {
-        var kmsConfig = new KMSConfiguration { KmsKeyArn = ExampleUtils.ExampleUtils.GetBranchKeyArn() };
-        // Create an AWS KMS Configuration to use with your KeyStore.
-        // The KMS Configuration MUST have the right access to the resources in the KeyStore.
-        var keystoreConfig = new KeyStoreConfig
+        // 1. Configure your KeyStore resource.
+        //    `ddbTableName` is the name you want for the DDB table that
+        //    will back your keystore.
+        //    `kmsKeyArn` is the KMS Key that will protect your branch keys and beacon keys
+        //    when they are stored in your DDB table.
+        var keyStoreConfig = new KeyStoreConfig
         {
-            // Client MUST have permissions to decrypt kmsConfig.KmsKeyArn
             KmsClient = new AmazonKeyManagementServiceClient(),
-            KmsConfiguration = kmsConfig,
+            KmsConfiguration = new KMSConfiguration{ KmsKeyArn = ExampleUtils.ExampleUtils.GetBranchKeyArn() },
             DdbTableName = ExampleUtils.ExampleUtils.GetKeyStoreName(),
             DdbClient = new AmazonDynamoDBClient(),
             LogicalKeyStoreName = ExampleUtils.ExampleUtils.GetLogicalKeyStoreName() 
         };
-        var materialProviders = new MaterialProviders(new MaterialProvidersConfig());
-        var keystore = new KeyStore(keystoreConfig);
+        var keyStore = new KeyStore(keyStoreConfig);
+        
+        // 2. Create the DynamoDb table that will store the branch keys and beacon keys.
+        //    This checks if the correct table already exists at `ddbTableName`
+        //    by using the DescribeTable API. If no table exists,
+        //    it will create one. If a table exists, it will verify
+        //    the table's configuration and will error if the configuration is incorrect.
+        //    It may take a couple minutes for the table to become ACTIVE,
+        //    at which point it is ready to store branch and beacon keys.
+        //    To create a KeyStore table, you need:
+        //     - "dynamodb:DescribeTable",
+        //     - "dynamodb:CreateTable",
+       
+        // This will NOT create a Key Store but check that the configured table name exists and it validates
+        // its construction. In order to check the construction is correct the configured IAM role
+        // MUST have `DescribeTable` permission on the Key Store resource.
```

Comment:
Suggestion: 
Semantic Line Breaks, grammar fixes, 
& tighter phrasing to remove the other comment.
```suggestion
        // Since AWS Crypto Tool's Example table already exists,
        // this will NOT create a Key Store
        // but check that the configured table name exists 
        // and validate the table's index configuration. 
        // In order to check this configuration,
        // the IAM role MUST have `DescribeTable` permission on 
        // the Key Store table.
```